### ISSUE TITLE
Misleading error message when implementer drops parameter default value

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#573]
```php
<?php

interface I
{
    public function method(bool $param = false): void;
}

class C implements I
{
    public function method(bool $param): void
    {

    }
}
```
Psalm says `C::method()` has more arguments, even though it only has more *required* arguments.

It's a fatal error in PHP.

**Comment by muglug** [comment]
Thanks!
